File: documentation-examples-java/src/java/dev/typr/foundations/docs/oracle/LobTypes.java
package dev.typr.foundations.docs.oracle;

import dev.typr.foundations.OracleType;
import dev.typr.foundations.OracleTypes;
import dev.typr.foundations.data.NonEmptyBlob;
import dev.typr.foundations.data.NonEmptyString;

@SuppressWarnings("unused")
public class LobTypes {
  // start
  OracleType<String> clobType = OracleTypes.clob;
  OracleType<String> nclobType = OracleTypes.nclob;
  OracleType<byte[]> blobType = OracleTypes.blob;

  // Non-empty variants
  OracleType<NonEmptyString> clobNonEmpty = OracleTypes.clobNonEmpty;
  OracleType<NonEmptyBlob> blobNonEmpty = OracleTypes.blobNonEmpty;
  // stop
}

File: documentation-examples-scala/src/scala/dev/typr/foundations/docs/oracle/LobTypes.scala
package dev.typr.foundations.docs.oracle

import dev.typr.foundations.{NonEmptyBlob, NonEmptyString, OracleType, OracleTypes}

@SuppressWarnings(Array("unused"))
object LobTypes:
  //start
  val clobType: OracleType[String] = OracleTypes.clob
  val nclobType: OracleType[String] = OracleTypes.nclob
  val blobType: OracleType[Array[Byte]] = OracleTypes.blob

  // Non-empty variants
  val clobNonEmpty: OracleType[NonEmptyString] = OracleTypes.clobNonEmpty
  val blobNonEmpty: OracleType[NonEmptyBlob] = OracleTypes.blobNonEmpty
  //stop
